homeassistant/components/zeroconf/__init__.py

```diff
@@ -2,93 +2,65 @@
 
 from __future__ import annotations
 
-import contextlib
 from contextlib import suppress
-from fnmatch import translate
-from functools import lru_cache, partial
+from functools import partial
 from ipaddress import IPv4Address, IPv6Address
 import logging
-import re
 import sys
-from typing import TYPE_CHECKING, Any, Final, cast
+from typing import Any, cast
 
 import voluptuous as vol
-from zeroconf import (
-    BadTypeInNameException,
-    InterfaceChoice,
-    IPVersion,
-    ServiceStateChange,
-)
-from zeroconf.asyncio import AsyncServiceBrowser, AsyncServiceInfo
+from zeroconf import InterfaceChoice, IPVersion
+from zeroconf.asyncio import AsyncServiceInfo
 
-from homeassistant import config_entries
 from homeassistant.components import network
 from homeassistant.const import (
     EVENT_HOMEASSISTANT_CLOSE,
     EVENT_HOMEASSISTANT_STOP,
     __version__,
 )
 from homeassistant.core import Event, HomeAssistant, callback
-from homeassistant.helpers import config_validation as cv, discovery_flow, instance_id
+from homeassistant.helpers import config_validation as cv, instance_id
 from homeassistant.helpers.deprecation import (
     DeprecatedConstant,
     all_with_deprecated_constants,
     check_if_deprecated_constant,
     dir_with_deprecated_constants,
 )
-from homeassistant.helpers.discovery_flow import DiscoveryKey
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.network import NoURLAvailableError, get_url
 from homeassistant.helpers.service_info.zeroconf import (
     ATTR_PROPERTIES_ID as _ATTR_PROPERTIES_ID,
     ZeroconfServiceInfo as _ZeroconfServiceInfo,
 )
 from homeassistant.helpers.typing import ConfigType
-from homeassistant.loader import (
-    HomeKitDiscoveredIntegration,
-    ZeroconfMatcher,
-    async_get_homekit,
-    async_get_zeroconf,
-    bind_hass,
-)
+from homeassistant.loader import async_get_homekit, async_get_zeroconf, bind_hass
 from homeassistant.setup import async_when_setup_or_start
 
+from .const import DOMAIN, ZEROCONF_TYPE
+from .discovery import (  # noqa: F401
+    DATA_DISCOVERY,
+    ZeroconfDiscovery,
+    build_homekit_model_lookups,
+    info_from_service,
+)
```

Comment:
Should we add this to `__all__` to make sure we don't break this API and make sure ruff stays sharper on the imports?